src/Cmdlets/GetKubeDeploymentCmdlet.cs

```diff
@@ -13,30 +13,47 @@ namespace Kubectl.Cmdlets {
     [Cmdlet(VerbsCommon.Get, "KubeDeployment")]
     [OutputType(new[] { typeof(DeploymentV1) })]
     public sealed class GetKubeDeploymentCmdlet : KubeApiCmdlet {
+        private const string DefaultParamSet = "DefaultParamSet";
+        private const string NamespaceObjectSet = "NamespaceObjectSet";
+
         [Parameter(ValueFromPipelineByPropertyName = true)]
         [Alias("Ns")]
         public string Namespace { get; set; } = "default";
 
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ParameterSetName = NamespaceObjectSet,
+            ValueFromPipeline = true)]
+        [ValidateNotNull()]
+        public NamespaceV1 NamespaceObject { get; set; }
+
         [Parameter()]
         public string LabelSelector { get; set; }
 
-        [Parameter(Position = 0, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true)]
+        [Parameter(
+            Position = 0,
+            ParameterSetName = DefaultParamSet,
```

Comment:
I think `-Name` and passing a namespace through pipeline input shouldn't be exclusive. `-Name` supports wildcards, so that can be useful to filter down the resources you want to get

Reply:
As I said in another comment, I think I'll just remove these changes from these specific cmdlets.